### PR TITLE
Added a personal high score to the UI

### PR DESCRIPTION
**Pull request by ghost**
## Description

My change just adds a personal high score to the UI
## Motivation and Context

This change removes a step for #129
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:

- [ ] My code follows the code style of this project. (I don't know. I think so)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: bot.user.js [hunk 1]
```diff
@@ -1076,6 +1076,23 @@ var userInterface = (function() {
                     'green;\">enabled' : 'red;\">disabled') +
                 '</span>';
         }
+        
+        // Set high score
+    setHighScore: function(){
```
Comment (ChadSki) [inline bot.user.js]: This error comes from a missing comma on line 1078.

This also threw off your IDE's indentation... line 1080 is correct but 1081 onward needs an extra four spaces.

**Comment by ChadSki** [comment]
Seems like a good idea! Testing now...

File: bot.user.js [hunk 2]
```diff
@@ -1078,21 +1078,21 @@ var userInterface = (function() {
         }
```
Comment (ChadSki) [inline bot.user.js]: You need a comma right here.

**Comment by ChadSki** [comment]
Where is the high score supposed to show up? I don't think I see it :confused:

**Comment by ghost** [comment]
@ChadSki should work now

**Comment by ghost** [comment]
and its meant to be above the fps

**Comment by ermiyaeskandary** [comment]
Please take note of what Hound has said.

**Comment by ghost** [comment]
ik 

> "w" is not defined.

is just bc its no running on slither

**Comment by ermiyaeskandary** [comment]
You have to use `window.` instead of `w` as they are global variables.

**Comment by ghost** [comment]
@ErmiyaEskandary just noticed that

File: bot.user.js [hunk 3]
```diff
@@ -1062,6 +1066,24 @@ var userInterface = (function() {
...
         },
 
+        // Set high score
+        setHighScore: function() {
+            var highScore = 0;
+            var scoreHUD = 0;
+            if (!window.snake || !window.fpsls || !window.fmlts) {
+                return;
+            }
+            var currentScore = Math.floor(150 * (window.fpsls[w.snake.sct] +
```
Comment (ermiyaeskandary) [inline bot.user.js]: Remove trailing spaces and replace `w.` with `window.` please.

File: bot.user.js [hunk 4]
```diff
@@ -1019,6 +1019,10 @@ var userInterface = (function() {
                     '</span>';
             }
 
+            if (window.playing) {
+                this.setHighScore()
```
Comment (ermiyaeskandary) [inline bot.user.js]: ^

**Comment by ermiyaeskandary** [comment]
@Classicanon Testing..... All violations of code style resolved. Great! But please have more meaningful commit titles as update bot.user.js doesn't tell us anything at a glance..

**Comment by ChadSki** [comment]
I still can't see the high score :(

**Comment by ermiyaeskandary** [comment]
Can't see it

**Comment by ermiyaeskandary** [comment]
It needs to be run in an existing interval if updated continuously and we don't use `this`. Refer to the wiki

**Comment by ermiyaeskandary** [comment]
Rejected

[Rejected with PullApprove](https://pullapprove.com/ErmiyaEskandary/Slither.io-bot/pull-request/167/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ErmiyaEskandary/Slither.io-bot)

File: bot.user.js [hunk 5]
```diff
@@ -1005,13 +1005,33 @@ var userInterface = (function() {
...
             // Botstatus overlay
             var generalStyle = '<span style = "opacity: 0.35";>';
             window.fps_overlay.innerHTML = generalStyle + 'FPS: ' +
                 userInterface.framesPerSecond.getFPS() + '</span>';
 
             if (window.position_overlay && window.playing) {
+                // Display Personal High Score
+                userInterface.setHighScore();
+                window.scoreHUD.innerHTML = generalStyle +
+                'Your High Score: ' + window.localStorage['highScoreLocal'];
```
Comment (ermiyaeskandary) [inline bot.user.js]: do `window.localstorage.highScoreLocal`

File: bot.user.js [hunk 6]
```diff
@@ -1005,13 +1005,33 @@ var userInterface = (function() {
...
+        setHighScore: function() {
+            var highScore = 1;
+            if (!window.snake || !window.fpsls || !window.fmlts) {
+                return;
+            }
+            var currentScore = Math.floor(150 * (window.fpsls[window.snake.sct] +
+            window.snake.fam / window.fmlts[window.snake.sct] - 1) - 50) / 10;
+            if (currentScore > highScore) {
+                highScore = currentScore;
+                window.localStorage['highScoreLocal'] = highScore
```
Comment (ermiyaeskandary) [inline bot.user.js]: Do `window.localstorage.highScoreLocal` 
Add a semicolon

**Comment by tjbcg** [comment]
Rejected too.

[Rejected with PullApprove](https://pullapprove.com/ErmiyaEskandary/Slither.io-bot/pull-request/167/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ErmiyaEskandary/Slither.io-bot)

**Comment by ermiyaeskandary** [comment]
It saves (?) but it doesn't load.

**Comment by ghost** [comment]
no it loads that's the only way for it to bridge over. it just isnt doing 
`if (currentScore>highScore) {
`
correctly

**Comment by ermiyaeskandary** [comment]
Rejected by 2 - closing. 
Self assigned.

[Rejected with PullApprove](https://pullapprove.com/ErmiyaEskandary/Slither.io-bot/pull-request/167/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ErmiyaEskandary/Slither.io-bot)

**Comment by ChadSki** [comment]
I like the idea but it needs to work properly and follow the code style.

If you want to take another stab at this, I'll be happy to review it again.

**Comment by ghost** [comment]
I'll probably try again but I need to figure out what is wrong

**Comment by tjbcg** [comment]
BTW, no real need for `if(currentScore > highScore)`. Just display `Math.max(currentScore, highScore)`.

**Comment by ghost** [comment]
Would it still save the variable?

**Pull request closed without merging**